service/history/transfer_queue_active_task_executor.go

```diff
@@ -817,12 +817,18 @@ func (t *transferQueueActiveTaskExecutor) processStartChildExecution(
 		targetNamespaceName = namespaceEntry.Name()
 	}
 
-	// copy version stamp from parent to child if:
-	// - command says to use compatible version
-	// - parent is using versioning
 	var sourceVersionStamp *commonpb.WorkerVersionStamp
+	var inheritedBuildId string
 	if attributes.InheritBuildId {
-		sourceVersionStamp = worker_versioning.StampIfUsingVersioning(mutableState.GetMostRecentWorkerVersionStamp())
+		inheritedBuildId = mutableState.GetAssignedBuildId()
```

Comment:
is this right? inherited and assigned are different concepts so far, but this is assigning one to the other? if this is intentional, it at least needs a comment to explain what's happening

Reply:
here we're setting inheritedBuildId of the child wf to the assignedBuildId of the parent. added a comment.

Reply:
oh whoops, I missed that this is for the child